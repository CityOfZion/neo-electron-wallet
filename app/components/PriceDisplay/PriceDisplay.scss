@import '../../styles/variables';

.container {
    font-size: .8em;
    opacity: 0.6;
}

.label {
    opacity: .6;
    margin-right: 4px;
}

.price {
    color: $thin-text-color;
    opacity: .8;
    cursor: pointer;
}

.neoPriceDisplay {
    padding-right: 38px
}