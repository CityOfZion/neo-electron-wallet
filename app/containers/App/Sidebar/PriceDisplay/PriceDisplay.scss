.navBarItem {
  font-size: .8em;
  color: #666;
}

.navBarItemLabel {
  color: #999;
  margin-right: 4px;
}
