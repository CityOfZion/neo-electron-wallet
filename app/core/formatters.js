// @flow
import { ASSETS } from './constants'
import numeral from 'numeral'

import { truncateNumber } from './math'
import numeral from 'numeral'

export const COIN_DECIMAL_LENGTH = 8
export const SHORT_DISPLAY_DECIMAL_LENGTH = 4

const formatWithDecimals = (value: number | string, decimals: number) => numeral(value).format(`0,0.${'0'.repeat(decimals)}`)

export const formatGAS = (value: number | string, shortDisplay: boolean = false): string => {
  const decimals = shortDisplay ? SHORT_DISPLAY_DECIMAL_LENGTH : COIN_DECIMAL_LENGTH
  const result = formatWithDecimals(truncateNumber(value, decimals), decimals)
  // https://github.com/adamwdraper/Numeral-js/issues/575
  if (result === 'NaN') {
    return toFixedDecimals(value, decimals)
  }
  return result
}

export const formatThousands = (value: number | string): string => numeral(value).format('0,0')

export const formatNEO = (value: number | string): string => formatThousands(value)

export const formatBalance = (symbol: SymbolType, balance: number, shortDisplay: boolean = false) => {
  if (symbol === ASSETS.NEO) {
    return formatNEO(balance)
  }
  return formatGAS(balance, shortDisplay)
}

export const toFixedDecimals = (value: number | string, decimals: number) => truncateNumber(parseFloat(value), decimals).toFixed(decimals)

export const formatFiat = (value: number | string): string => numeral(value).format('0,0.00')
