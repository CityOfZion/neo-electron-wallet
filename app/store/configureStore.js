import { createStore, applyMiddleware } from 'redux';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';
// export type GlobalState = {}
export var INITIAL_STATE = {};
var loggerMiddleware = createLogger();
function configureStore(initialState) {
    if (initialState === void 0) { initialState = INITIAL_STATE; }
    return createStore(rootReducer, initialState, process.env.NODE_ENV === 'production'
        ? applyMiddleware(thunk)
        : applyMiddleware(thunk, loggerMiddleware));
}
export var store = configureStore();
