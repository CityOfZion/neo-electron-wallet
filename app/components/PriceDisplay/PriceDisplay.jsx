// @flow
import React from 'react'
import { formatFiat } from '../../core/formatters'
import styles from './PriceDisplay.scss'
import classNames from 'classnames'
import { CURRENCIES } from '../../core/constants'

type Props = {
  neoPrice: number,
  gasPrice: number,
  currencyCode: string
}

const PriceDisplay = ({ neoPrice, gasPrice, currencyCode }: Props) => {
  let neoDisplayPrice = '-'
  let gasDisplayPrice = '-'
  const currencySymbol = CURRENCIES[currencyCode].symbol

  if (neoPrice) {
    neoDisplayPrice = formatFiat(neoPrice)
  }

  if (gasPrice) {
    gasDisplayPrice = formatFiat(gasPrice)
  }

  return (
    <div className={styles.container}>
      <span id='neoPrice' className={styles.neoPriceDisplay}>
        <span className={styles.label}>NEO</span>
        <span className={classNames('price', styles.price)}>{currencySymbol}{neoDisplayPrice}</span>
      </span>
      <span id='gasPrice'>
        <span className={styles.label}>GAS</span>
        <span className={classNames('price', styles.price)}>{currencySymbol}{gasDisplayPrice}</span>
      </span>
    </div>
  )
}

export default PriceDisplay
