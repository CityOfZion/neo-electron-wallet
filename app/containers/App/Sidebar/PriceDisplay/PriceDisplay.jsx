// @flow
import React from 'react'

import { formatFiat } from '../../../../core/formatters'
import { CURRENCIES } from '../../../../core/constants'

import styles from './PriceDisplay.scss'

type Props = {
  neoPrice: number,
  gasPrice: number,
  currencyCode: string
}

const PriceDisplay = ({ neoPrice, gasPrice, currencyCode }: Props) => {
  const neoDisplayPrice = neoPrice ? formatFiat(neoPrice) : '-'
  const gasDisplayPrice = gasPrice ? formatFiat(gasPrice) : '-'
  const currencySymbol = CURRENCIES[currencyCode].symbol

  return [
    <div key='neoPriceDisplay' id='neoPrice' className={styles.navBarItem}>
      <span className={styles.navBarItemLabel}>NEO</span>
      <span className='price'>{currencySymbol}{neoDisplayPrice}</span>
    </div>,
    <div key='gasPriceDisplay' id='gasPrice' className={styles.navBarItem}>
      <span className={styles.navBarItemLabel}>GAS</span>
      <span className='price'>{currencySymbol}{gasDisplayPrice}</span>
    </div>
  ]
}

export default PriceDisplay
