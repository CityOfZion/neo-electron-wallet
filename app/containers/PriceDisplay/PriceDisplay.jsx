// @flow
import React, { Component } from 'react'
import { formatFiat } from '../../core/formatters'
import styles from './PriceDisplay.scss'
import classNames from 'classnames'

type Props = {
  neoPrice: number,
  gasPrice: number
}

export default class PriceDisplay extends Component<Props> {
  render () {
    const { neoPrice, gasPrice } = this.props

    let neoDisplayPrice = '-'
    let gasDisplayPrice = '-'

    if (neoPrice) {
      neoDisplayPrice = formatFiat(neoPrice)
    }

    if (gasPrice) {
      gasDisplayPrice = formatFiat(gasPrice)
    }

    return (
      <div className={styles.container}>
        <span id='neoPrice' className={styles.neoPriceDisplay}>
          <span className={styles.label}>NEO</span>
          <span className={classNames('price', styles.price)}>${neoDisplayPrice}</span>
        </span>
        <span id='gasPrice'>
          <span className={styles.label}>GAS</span>
          <span className={classNames('price', styles.price)}>${gasDisplayPrice}</span>
        </span>
      </div>
    )
  }
}
