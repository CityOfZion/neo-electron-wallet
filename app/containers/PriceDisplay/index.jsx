// @flow
import { connect } from 'react-redux'
import PriceDisplay from './PriceDisplay'
import { getNeoPrice, getGasPrice } from '../../modules/wallet'

const mapStateToProps = (state) => ({
  neoPrice: getNeoPrice(state),
  gasPrice: getGasPrice(state)
})

export default connect(mapStateToProps)(PriceDisplay)
