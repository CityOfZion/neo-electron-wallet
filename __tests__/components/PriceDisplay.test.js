import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { mount, shallow } from 'enzyme'
import PriceDisplay from '../../app/containers/PriceDisplay'
import { formatFiat } from '../../app/core/formatters'

const initialState = {
  wallet: {
    neoPrice: 25.48,
    gasPrice: 18.10
  }
}

const setup = (state = initialState, shallowRender = true) => {
  const store = configureStore([thunk])(state)

  let wrapper
  if (shallowRender) {
    wrapper = shallow(<PriceDisplay store={store} />)
  } else {
    wrapper = mount(
      <Provider store={store}>
        <PriceDisplay />
      </Provider>
    )
  }

  return {
    store,
    wrapper
  }
}

describe('PriceDisplay', () => {
  test('renders without crashing', (done) => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
    done()
  })
  test('correctly renders data from state', (done) => {
    const { wrapper } = setup(initialState, false)

    const neoPrice = wrapper.find('#neoPrice .price')
    const gasPrice = wrapper.find('#gasPrice .price')
    const expectedNeoPrice = formatFiat(initialState.wallet.neoPrice)
    const expectedGasPrice = formatFiat(initialState.wallet.gasPrice)

    expect(neoPrice.text()).toEqual(`$${expectedNeoPrice}`)
    expect(gasPrice.text()).toEqual(`$${expectedGasPrice}`)
    done()
  })
})
