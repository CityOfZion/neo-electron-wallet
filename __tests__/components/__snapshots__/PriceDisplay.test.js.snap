// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDisplay renders without crashing 1`] = `
<PriceDisplay
  dispatch={[Function]}
  gasPrice={18.1}
  neoPrice={25.48}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
